olp-cpp-sdk-core/src/cache/DefaultCache.cpp

```diff
@@ -281,7 +288,17 @@ DefaultCache::StorageOpenResult DefaultCache::SetupStorage() {
   if (settings_.max_memory_cache_size > 0) {
     memory_cache_.reset(new InMemoryCache(settings_.max_memory_cache_size));
   }
-  if (settings_.disk_path) {
+
+  // Temporary code for backwards compatibility.
+  if (!settings_.disk_path_mutable) {
+    PORTING_PUSH_WARNINGS()
+    PORTING_CLANG_GCC_DISABLE_WARNING("-Wdeprecated-declarations")
+    PORTING_MSVC_DISABLE_WARNINGS(4996)
+    settings_.disk_path_mutable = settings_.disk_path;
+    PORTING_POP_WARNINGS()
+  }
```

Comment:
Please move to an anonymous helper so that we avoid this in the middle of the code.
Something like ValidateDiskPath(settings_).